pkg/plugins/v1alpha1/plugin/interface.go

```diff
@@ -0,0 +1,21 @@
+/*
+Copyright © 2021 Aspect Build Systems Inc
+
+Not licensed for re-use.
+*/
+
+package plugin
+
+import (
+	buildeventstream "aspect.build/cli/bazel/buildeventstream/proto"
+	"aspect.build/cli/pkg/ioutils"
+)
+
+// Plugin determines how an aspect Plugin should be implemented.
```

Comment:
this is handy for plugins written in Go, should we mention that it's optional for plugins in other languages?

also, why isn't this produced in the protoc output .pb.go file instead?

Reply:
I think we can mention it in the SDK doc.

As for your second question, I wanted to abstract the internal RPC communication so that the user doesn't need to care whether it's gRPC or not. It doesn't feel like the user of the SDK is doing gRPC, it's just implementing a simple Go interface.